markdown/events/2021/bytesize-18-dev-envs-workflows.md

```diff
@@ -29,3 +29,40 @@ This will cover:
+
+[4:52](https://youtu.be/OF55x-FT5WE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=292) To follow what was presented in [Bytesize#11](https://nf-co.re/events/2021/bytesize-11-dev-envs-workflows), I’ll cover how to set up your own fork on GitHub, configure your local clone, work on other people’s code and pull in updates.
+
+[5:09](https://youtu.be/OF55x-FT5WE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=309) I use the GitHub command line. Let’s say that I’ve forked the pipeline. What I usually do is to append the name of the original directory to my fork, so that I can trace it back if I need to. If I’d like to check the code, I use the GitHub command line interface. I copy the path and go to my workspace folder and then clone it. An advantage of using the GitHub command line interface is that the remote is already configured. This makes it easy.
+
+[6:26](https://youtu.be/OF55x-FT5WE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=386) Then if I’m doing some actual work. I use `zsh` on my terminal so that I know which branch I am on (I’m on `dev` here). When I start working on a new feature, I do `git checkout -b` on the branch I am working on and then specify my new feature. This takes me to that branch, and I work on my branch where I’m now going to modify a file. Let’s head to VSCode, take a look at the pipeline, then head to the CHANGELOG.md, and make some small changes. I’m happy with the modification. So now I go back to the command line and say `git add` followed by the name of the file. Then I commit with the message `git commit -m` and add the feature (“feat:”) to update the file name (CHANGELOG.md in this case). I specify “feat:” if I’ve done something, “fix:” if I’ve fixed something, “chores:” if I had to do something. Then if I want to push, I use `git push -u origin` with the name of the branch. So that’s it. My branch should now be available on GitHub. You can also automatically create your PR in the GitHub command line interface with `gh pr create`. That’s quite fun and it’s really useful to make a PR like I needed to do when I updated the social media images for the different pipelines. You need to specify the base branch `--base dev`, a title `--title` for example “test”, and you can specify a `--body` for the message for example “testing something, do not merge”. When I do that, it asks me where I’d like to make the repository, and I add that information.
```

Comment:
```suggestion
[6:26](https://youtu.be/OF55x-FT5WE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=386) Then if I’m doing some actual work. I use `zsh` on my terminal so that I know which branch I am on (I’m on `dev` here). When I start working on a new feature, I do `git checkout -b` on the branch I am working on and then specify my new feature. This takes me to that branch, and I work on my branch where I’m now going to modify a file. Let’s head to VSCode, take a look at the pipeline, then head to the CHANGELOG.md, and make some small changes. I’m happy with the modification. So now I go back to the command line and say `git add` followed by the name of the file. Then I commit with the message `git commit -m` and add the feature (“feat:”) to update the file name (`CHANGELOG.md` in this case). I specify “feat:” if I’ve done something, “fix:” if I’ve fixed something, “chores:” if I had to do something. Then if I want to push, I use `git push -u origin` with the name of the branch. So that’s it. My branch should now be available on GitHub. You can also automatically create your PR in the GitHub command line interface with `gh pr create`. That’s quite fun and it’s really useful to make a PR like I needed to do when I updated the social media images for the different pipelines. You need to specify the base branch `--base dev`, a title `--title` for example “test”, and you can specify a `--body` for the message for example “testing something, do not merge”. When I do that, it asks me where I’d like to make the repository, and I add that information.
```